firefox/background.entry.js

```diff
@@ -69,9 +80,17 @@ const {
 
 addCommonBackgroundListeners(addListener);
 
-addListener('deleteCookies', cookies =>
-	cookies.forEach(({ name }) => cookieManager.remove('.reddit.com', name, '/', false, false))
-);
+addListener('deleteCookies', cookies => {
+	for (const { host, name, path, originAttributes } of cookieManager.enumerator::asGenerator(Ci.nsICookie)) {
```

Comment:
I'd love to be able to get the `originAttributes` without iterating through every single cookie, but I can't find any (documented) API that will let me do that (in [nsICookieManager](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsICookieManager) or [nsICookieManager2](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsICookieManager2)).

One might expect to be able to use [getCookiesFromHost()](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsICookieManager2#getCookiesFromHost()), but nope, it also requires `originAttributes`, though this is not documented:

![](https://cloud.githubusercontent.com/assets/7673145/17349370/1b240184-58eb-11e6-9502-1f4e3363d6b3.PNG)

In any case, we'll be switching to WebExtensions at some point in the (hopefully) near future and this code will disappear.